heat/core/arithmetics.py

```diff
@@ -129,6 +169,65 @@ def _add(self, other):
 DNDarray.__radd__.__doc__ = add.__doc__
 
 
+def add_(t1: DNDarray, t2: Union[DNDarray, float]) -> DNDarray:
+    """
+    Element-wise in-place addition of values of two operands.
+    Takes the first operand (:class:`~heat.core.dndarray.DNDarray`) and element-wise adds the
+    element(s) of the second operand (scalar or :class:`~heat.core.dndarray.DNDarray`) in-place,
+    i.e. the elements of ``t1`` are overwritten by the results of element-wise addition of ``t1``
+    and ``t2``.
+
+    Parameters
+    ----------
+    t1: DNDarray
+        The first operand involved in the addition
+    t2: DNDarray or scalar
+        The second operand involved in the addition
+
+    Examples
+    --------
+    >>> import heat as ht
+    >>> T1 = ht.float32([[1, 2], [3, 4]])
+    >>> T2 = ht.float32([[2, 2], [2, 2]])
+    >>> ht.add_(T1, T2)
+    DNDarray([[3., 4.],
+              [5., 6.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> T1
+    DNDarray([[3., 4.],
+              [5., 6.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> T2
+    DNDarray([[2., 2.],
+              [2., 2.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> s = 2.0
+    >>> ht.add_(T2, s)
+    DNDarray([[4., 4.],
+              [4., 4.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> T2
+    DNDarray([[4., 4.],
+              [4., 4.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> s
+    2.0
+    """
+    if not isinstance(t1, DNDarray):
+        raise TypeError(
+            "Input 1 must be a DNDarray and input 2 either a DNDarray or a scalar. "
+            + "But your inputs were from "
+            + str(type(t1))
+            + " and "
+            + str(type(t2))
+            + "."
+        )
+
+    def wrap_add_(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
+        return a.add_(b)
+
+    return _operations.__binary_op(wrap_add_, t1, t2)
+
+
+DNDarray.__iadd__ = lambda self, other: add_(self, other)
+DNDarray.__iadd__.__doc__ = add_.__doc__
```

Comment:
```suggestion
DNDarray.__iadd__ = add_
```
You don't really need the lambda function here and the reference to the docstring then.